clintrials/core/simulation.py

```
@@ -316,120 +190,92 @@ def run(self, mode="iterative", n_sims=1, cohort_size=1, show_progress=False, **
-
             design.reset()
             if recruitment_stream:
                 recruitment_stream.reset()
-
             i = 0
-            # Some protocols might not have max_size
-            max_size = getattr(design, "max_size", lambda: float('inf'))()
+            max_size = getattr(design, 'max_size', lambda: float('inf'))()
+            if max_size == float('inf'):
+                # Require an explicit bound if the protocol is not self-terminating
+                if not hasattr(design, 'has_more'):
```

Ruff (0.15.21)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)